### PR TITLE
fix: pod annotated by missing patches should fail

### PR DESCRIPTION
**Pull request by bartsmykla**
### Summary

It reverts past changes which made workloads (pods) annotated with
some container patches which were missing to succeed kuma injection.

Now we will log the "pretty" error and fail the injection.

ref. https://github.com/kumahq/kuma/pull/4362, https://github.com/kumahq/kuma/issues/4356

### Full changelog

n/a

### Issues resolved

n/a

### Documentation

- [ ] ~~Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)~~

### Testing

- [x] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [x] Manual testing on Kubernetes

### Backwards compatibility

- [ ] ~~Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.~~
- [x] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)

### CODE REVIEW

File: pkg/plugins/runtime/k8s/webhooks/injector/injector_test.go [hunk 1]
```diff
@@ -630,21 +630,5 @@ spec:
...
-                name: default`,
-			namespace: `
-              apiVersion: v1
-              kind: Namespace
-              metadata:
-                name: default
-                annotations:
-                  kuma.io/sidecar-injection: enabled`,
-			cfgFile: "inject.config.yaml",
-		}),
```
Comment (lahabana) [inline pkg/plugins/runtime/k8s/webhooks/injector/injector_test.go:648]: Should this be replaced by a test to check there's an error?

Comment (bartsmykla) [reply pkg/plugins/runtime/k8s/webhooks/injector/injector_test.go:648]: I'm exactly working on this as we speak :-D

Comment (bartsmykla) [reply pkg/plugins/runtime/k8s/webhooks/injector/injector_test.go:648]: done

File: pkg/plugins/runtime/k8s/webhooks/injector/injector.go [hunk 2]
```diff
@@ -256,10 +256,12 @@ func (i *KumaInjector) loadContainerPatches(
 		}
 	}
 
+	var missingPatches []string
+
 	for _, patchName := range patchNames {
 		containerPatch := &mesh_k8s.ContainerPatch{}
 		if err := i.client.Get(ctx, kube_types.NamespacedName{Namespace: i.systemNamespace, Name: patchName}, containerPatch); err != nil {
-			logger.Error(err, "could not get a ContainerPatch")
+			missingPatches = append(missingPatches, patchName)
```
Comment (lobkovilya) [inline pkg/plugins/runtime/k8s/webhooks/injector/injector.go:264]: Should we check if `err` is NotFoundError?

Comment (bartsmykla) [reply pkg/plugins/runtime/k8s/webhooks/injector/injector.go]: addressed

**Review by lahabana — APPROVED** [review]

**Pull request merged**